symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/stream/StreamService.java

```diff
@@ -0,0 +1,274 @@
+package com.symphony.bdk.core.service.stream;
+
+import com.symphony.bdk.core.api.invoker.ApiException;
+import com.symphony.bdk.core.api.invoker.ApiRuntimeException;
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.core.service.stream.constant.AttachmentSort;
+import com.symphony.bdk.gen.api.StreamsApi;
+
+import com.symphony.bdk.gen.api.model.RoomDetail;
+import com.symphony.bdk.gen.api.model.Stream;
+
+import com.symphony.bdk.gen.api.model.StreamAttachmentItem;
+import com.symphony.bdk.gen.api.model.StreamAttributes;
+import com.symphony.bdk.gen.api.model.StreamFilter;
+import com.symphony.bdk.gen.api.model.V2AdminStreamFilter;
+import com.symphony.bdk.gen.api.model.V2AdminStreamList;
+import com.symphony.bdk.gen.api.model.V2MembershipList;
+import com.symphony.bdk.gen.api.model.V2RoomSearchCriteria;
+import com.symphony.bdk.gen.api.model.V2StreamAttributes;
+import com.symphony.bdk.gen.api.model.V3RoomAttributes;
+
+import com.symphony.bdk.gen.api.model.V3RoomDetail;
+
+import com.symphony.bdk.gen.api.model.V3RoomSearchResults;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+
+/**
+ * Service class for managing streams.
+ *
+ * This service is used for retrieving information about a particular stream or
+ * chatroom, searching streams, listing members, attachments of a particular stream,
+ * perform some action related to a stream like:
+ * <p><ul>
+ *   <li>Create a IM or MIM</li>
+ *   <li>Create a chatroom</li>
+ *   <li>Activate or Deactivate a chatroom</li>
+ *   <li></li>
+ * </ul></p>
+ *
+ */
+@Slf4j
+public class StreamService {
+
+  private final StreamsApi streamsApi;
+  private final AuthSession authSession;
+
+  public StreamService(StreamsApi streamsApi, AuthSession authSession) {
+    this.streamsApi = streamsApi;
+    this.authSession = authSession;
+  }
+
+  /**
+   * Create a new single or multi party instant message conversation between the caller and specified users.
+   *
+   * The caller is implicitly included in the members of the created chat.
+   *
+   * Duplicate users will be included in the membership of the chat but
+   * the duplication will be silently ignored.
+   *
+   * If there is an existing IM conversation with the same set of participants then
+   * the id of that existing stream will be returned.
+   *
+   * @param uids List of user ids of the participants
+   * @return The created IM or MIM
+   * @see <a href="https://developers.symphony.com/restapi/reference#create-im-or-mim">Create IM or MIM</a>
+   */
+  public Stream createIMorMIM(List<Long> uids) {
```

Comment:
I'd rename this method to `create` with several prototypes: 
- `create(Long)` for creating an IM
- `create(List<Long>)` or `create(Long...)` for creating MIM
- `create(V3RoomAttributes)` for creating a room